src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseIntegrationTestService.java

```diff
@@ -1658,11 +1660,14 @@ void unlockAllRepositories() throws Exception {
         final var endpoint = ProgrammingExerciseResourceEndpoints.UNLOCK_ALL_REPOSITORIES.replace("{exerciseId}", String.valueOf(programmingExercise.getId()));
         request.put(ROOT + endpoint, null, HttpStatus.OK);
 
-        verify(versionControlService).configureRepository(programmingExercise, participation1, true);
-        verify(versionControlService).configureRepository(programmingExercise, participation2, true);
+        verify(versionControlService, timeout(300)).configureRepository(programmingExercise, participation1, true);
+        verify(versionControlService, timeout(300)).configureRepository(programmingExercise, participation2, true);
 
         userUtilService.changeUser(userPrefix + "instructor1");
 
+        // Wait until the notifications got created
+        Thread.sleep(300);
```

Comment:
Is this automatically "speed-improved" like it is in our client tests? Otherwise, won't this slow down the server tests quite a bit?

Reply:
No, it always waits 300ms. I'll try to replace it with a Await.until() call.

Reply:
Ok 👍 I also don't have another idea on how to do this. I suggest we leave this thread open in case someone comes across this and has a better idea. Otherwise I'm also fine with this.